sql/core/src/main/scala/org/apache/spark/sql/execution/command/views.scala

```diff
@@ -259,7 +259,10 @@ case class AlterViewAsCommand(
   def markAsAnalyzed(): LogicalPlan = copy(isAnalyzed = true)
 
   override def run(session: SparkSession): Seq[Row] = {
-    if (session.sessionState.catalog.isTempView(name)) {
+    val isTemporary = session.sessionState.catalog.isTempView(name)
+    verifyTemporaryObjectsNotExists(session.sessionState.catalog, isTemporary, name, query)
```

Comment:
let's remove `verifyAutoGeneratedAliasesNotExists(analyzedPlan, isTemporary = false, name)` in `alterTemporaryView ` function

Reply:
OK